neptune/alpha/handler.py

```diff
@@ -73,26 +75,39 @@ def save_files(self, value: Union[str, Iterable[str]], wait: bool = False) -> No
             else:
                 attr.save_files(value, wait)
 
-    def log(self, value: Union[int, float, str, Image], step=None, timestamp=None, wait: bool = False) -> None:
-        verify_type("value", value, (int, float, str, Image))
+    def log(self,
+            value: Union[int, float, str, Image, Iterable[int], Iterable[float], Iterable[str], Iterable[Image]],
+            step=None,
+            timestamp=None,
+            wait: bool = False) -> None:
+        verify_type("value", value, (int, float, str, Image, Iterable))
         verify_type("step", step, (int, float, type(None)))
         verify_type("timestamp", step, (int, float, type(None)))
+
         with self._experiment.lock():
             attr = self._experiment.get_attribute(self._path)
             if not attr:
-                if isinstance(value, (float, int)):
+                if is_collection(value):
+                    if value:
+                        first_value = first(value)
+                    else:
+                        raise ValueError("Cannot deduce value type: `value` cannot be empty")
+                else:
+                    first_value = value
+
+                if isinstance(first_value, (float, int)):
                     attr = FloatSeries(self._experiment, parse_path(self._path))
-                elif isinstance(value, str):
+                elif isinstance(first_value, str):
                     attr = StringSeries(self._experiment, parse_path(self._path))
-                elif isinstance(value, Image):
+                elif isinstance(first_value, Image):
                     attr = ImageSeries(self._experiment, parse_path(self._path))
                 attr.log(value, step=step, timestamp=timestamp, wait=wait)
                 self._experiment.set_attribute(self._path, attr)
             else:
                 attr.log(value, step=step, timestamp=timestamp, wait=wait)
 
-    def add(self, values: Iterable[str], wait: bool = False) -> None:
-        verify_collection_type("values", values, str)
+    def add(self, values: Union[str, Iterable[str]], wait: bool = False) -> None:
+        verify_type("values", values, (str, Iterable))
```

Comment:
verify_collection_type should be called for a collection

Reply:
It is verified in attr.add()